core/templates/dev/head/pages/exploration_editor/translation_tab/AudioTranslationBarDirective.js

```diff
@@ -93,6 +95,10 @@ oppia.directive('audioTranslationBar', [
           $scope.checkingMicrophonePermission = false;
           $scope.audioTimerIsShown = true;
           $scope.audioIsCurrentlyBeingSaved = false;
+          $scope.translationIsDisallowed = !GLOBALS.can_translate;
+          $scope.userIsGuest = !GLOBALS.userIsLoggedIn;
+          $scope.dropAreaIsShown = !$scope.userIsGuest &&
```

Comment:
Add parens, and break after '(' (see my previous comment).

Also you have "isShown" here and "isAccessible" elsewhere, should this be combined into the same thing? We keep repeating the logic in many places and it would be better to just define the logic once if possible.

Reply:
I'm going to answer both of the question here because separate answers would just over-complicate things.

Would it be a good idea to use the GLOBALS on the html template then to simplify this?

Reply:
What do you mean by "use the GLOBALS on the html template"?

Reply:
Instead of creating scope variables I would use the GLOBALS.can_translate in the template for ng-if example.

Reply:
That probably won't work, because GLOBALS is not defined on scope. When you use a variable in a template it needs to be one that's defined on the scope.

Reply:
I see then if I simplify the logic to one place like the following
`
scope.dropAreaIsAccessible = (GLOBALS.can_translate &&
  GLOBALS.userIsLoggedIn);`

This will work except that the message when the user is a guest will never show.

Reply:
Just to be clear, I don't mean changing the intended user-facing behaviour. What I meant is that we should just do this "computation" a single time in the code, store the result, and reuse it after that rather than recomputing it, so that there's a single source of truth. But this shouldn't necessitate a change in the messages that are shown and when they're shown etc.

Reply:
Yeah I see what you are saying, so in this case $scope.dropAreaIsShown be unnecessary then because we already have $scope.translationIsDisallowed  and $scope.userIsGuest defined.

Reply:
Yep, though the reasoning is actually more like "dropAreaIsShown" is unnecessary because you already have "dropAreaIsAccessible". I.e. you just need one variable to refer to.

Reply:
Ok will remove dropAreaIsShown. The dropAreaIsAccessible is the main boolean toggle.